src/utils.ts

```diff
@@ -155,7 +155,9 @@ export const execSync = (command: string) =>
 export const getOptionalInput = (name: string) => getInput(name) || undefined
 
 export const getUsername = async (api: Gitlab) => {
-  return process.env.GITLAB_CI_USER_NAME == null
-    ? await api.Users.current().then(currentUser => currentUser.username)
-    : process.env.GITLAB_CI_USER_NAME
+  if (process.env.GITLAB_CI_USER_NAME != null) {
+    return process.env.GITLAB_CI_USER_NAME
+  }
+  const currentUser = await api.Users.showCurrentUser()
+  return currentUser.username
```

Comment:
```suggestion
  return process.env.GITLAB_CI_USER_NAME || await api.Users.showCurrentUser().then(currentUser => currentUser.username)
```

I believe it's much cleaner personally.

Reply:
I believe that we do not need the `await` here since promises are automatically flattened. Am I correct?